### ISSUE TITLE
Numpy < 2 causes TypeError in IOU Matcher with uint64 labels

### ISSUE THREAD

**Issue by cmalinmayor** [Janelia-Trackathon-2023/traccuracy#152]
* traccuracy version: f493cff6c3b2e2b52f786c29ea13a6a63dc31d8a (check_empty_subgraph branch)
* Python version: 3.10.14
* Numpy version: 1.26.4
* Operating System: MacOS

# Description

# Minimal example to reproduce the bug
pip install numpy<2
add  ```y1 = y1.astype("uint64")``` to test_iou_matcher.test__match_nodes before matching (line 41)
pytest tests/matchers/test_iou_matcher.py

Traceback:
```
    def _match_nodes(gt, res, threshold=0.5, one_to_one=False):
        """Identify overlapping objects according to IoU and a threshold for minimum overlap.

        QUESTION: Does this rely on sequential segmentation labels

        Args:
            gt (np.ndarray): labeled frame
            res (np.ndarray): labeled frame
            threshold (optional, float): threshold value for IoU to count as same cell. Default 1.
                If segmentations are identical, 1 works well.
                For imperfect segmentations try 0.6-0.8 to get better matching
            one_to_one (optional, bool): If True, forces the mapping to be one-to-one by running
                linear assignment on the thresholded iou array. Default False.

        Returns:
            gtcells (np arr): Array of overlapping ids in the gt frame.
            rescells (np arr): Array of overlapping ids in the res frame.
        """
        print("Gt dtype: ", gt.dtype, " result dtype ", res.dtype)
        print("Gt max dtype: ", np.max(gt).dtype, " result max dtype ", np.max(res).dtype)
        print("Gt max: ", np.max(gt), " result max ", np.max(res))

        print("Gt max +1 dtype: ", (np.max(gt) + 1).dtype, " result max + 1 dtype ", (np.max(res) + 1).dtype)
        print("Gt max +1: ", np.max(gt) + 1, " result max + 1", np.max(res) + 1)

>       iou = np.zeros((np.max(gt) + 1, np.max(res) + 1))
E       TypeError: 'numpy.float64' object cannot be interpreted as an integer

src/traccuracy/matchers/_iou.py:40: TypeError
```

The print statements print: 
```
Gt dtype:  uint64    result dtype:  uint64
Gt max dtype:  uint64   result max dtype:  uint64
Gt max:  5      result max:  5
Gt max +1 dtype:  float64       result max +1 dtype:  float64
Gt max +1:  6.0  result max +1:  6.0
```
So it appears that adding 1 to a np.uint64 changes the dtype to float? I verified that it works fine with (numpy=2.0, dtype=np.uint64) as well as (numpy<2, dtype=np.uint32).
@msschwartz21 Should we just cast to to int to cover this weird edge case?

# Severity
- [ ] Unusable
- [x] Annoying, but still functional (if you can massage your dependencies or dtypes)
- [ ] Very minor

**Comment by msschwartz21** [comment]
This is such a weird edge case, but I agree let's just cast that to int and add that fix to #153. Does the +/- 1 issue come up again anywhere else?

**Comment by cmalinmayor** [comment]
Once I cast to int, nothing else errors.... so I assume it's the only place! Adding to the PR now
